executor/src/witgen/bit_constraints.rs

```diff
@@ -396,53 +395,49 @@ namespace Global(2**20);
     { D } in { SHIFTED };
 ";
         let analyzed = pil_analyzer::analyze_string(pil_source);
-        let (constants, degree) = crate::constant_evaluator::generate(&analyzed);
-        let mut known_constraints = constants
+        let (constants, _) = crate::constant_evaluator::generate(&analyzed);
+        let fixed_polys = constants
             .iter()
-            .filter_map(|(name, values)| {
-                process_fixed_column(values).map(|(constraint, _full)| (*name, constraint))
+            .enumerate()
+            .map(|(i, (name, _))| PolynomialReference {
+                name: name.to_string(),
+                poly_id: Some(PolyID {
+                    id: i as u64,
+                    ptype: pil_analyzer::PolynomialType::Constant,
+                }),
+                index: None,
+                next: false,
+            })
+            .collect::<Vec<_>>();
+        let mut known_constraints = fixed_polys
+            .iter()
+            .zip(&constants)
+            .filter_map(|(poly, (_, values))| {
+                process_fixed_column(values).map(|(constraint, _full)| (poly, constraint))
             })
             .collect::<BTreeMap<_, _>>();
         assert_eq!(
-            known_constraints,
+            known_constraints
+                .iter()
+                .map(convert_constraints)
+                .collect::<BTreeMap<_, _>>(),
             vec![
                 ("Global.BYTE", BitConstraint::from_max_bit(7)),
                 ("Global.BYTE2", BitConstraint::from_max_bit(15)),
                 ("Global.SHIFTED", BitConstraint::from_mask(0xff0)),
             ]
             .into_iter()
-            .collect()
-        );
-        // TODO write some test code to generate FixedData directly from `analyzed`
-        let witness_cols: Vec<WitnessColumn> = analyzed
```

Comment:
Where's this gone?

Reply:
It's all not needed any more, because `FixedData` was removed from the params of `propagate_constraints`.